uwsgi_tools/curl.py

```diff
@@ -45,17 +42,34 @@ def curl(uwsgi_addr, url, method='GET', body='', timeout=0, headers=(),
     else:
         port = None
 
+    if body is None:
+        body = ''.encode('utf-8')
+    else:
+        body = body.encode('utf-8')
+
     var = {
         'SERVER_PROTOCOL': 'HTTP/1.1',
         'PATH_INFO': parts_uri.path,
         'REQUEST_METHOD': method.upper(),
         'REQUEST_URI': uri,
         'QUERY_STRING': parts_uri.query,
         'HTTP_HOST': host,
+        'CONTENT_LENGTH': str(len(body)),
+        # Other varaibles seen in nginx's uwsgi_params file but not explicitly
+        # handled anywhere in this file
+        # https://github.com/nginx/nginx/blob/master/conf/uwsgi_params
+        # DOCUMENT_ROOT
+        # REQUEST_SCHEME
+        # HTTPS
+        # REMOTE_ADDR
+        # REMOTE_PORT
     }
+
     for header in headers or ():
         key, _, value = header.partition(':')
-        var['HTTP_' + key.strip().upper()] = value.strip()
+        var['HTTP_' + key.strip().upper().replace('-', '_')] = value.strip()
+    if 'HTTP_CONTENT-TYPE' in var.keys():
```

Comment:
It's probably a typo there and I wonder do we even need `HTTP_CONTENT_TYPE` there at all?

Reply:
This is indeed a typo and should be `HTTP_CONTENT_TYPE`.

However, we do still need it. In much the same way we're special casing `SERVER_NAME` and `SERVER_PORT` we also need to extract the well used uWSGI param `CONTENT_TYPE` from the header `HTTP_CONTENT_TYPE`.